lib/cache/cache.go

```diff
@@ -363,6 +363,19 @@ func ForDiscovery(cfg Config) Config {
 	return cfg
 }
 
+// ForOkta sets up watch configuration for Okta servers.
+func ForOkta(cfg Config) Config {
+	cfg.target = "okta"
+	cfg.Watches = []types.WatchKind{
+		{Kind: types.KindUser},
+		{Kind: types.KindAccessRequest},
+		{Kind: types.KindOktaImportRule},
+		{Kind: types.KindOktaAssignment},
+	}
+	cfg.QueueSize = defaults.DiscoveryQueueSize
+	return cfg
+}
```

Comment:
Caches cannot watch resources that are not present in their upstream cache.  This means that you need to add the import rules and assignments to the auth cache in order for them to be propagated to this one.